boxsdk/auth/oauth2.py

```diff
@@ -343,6 +354,57 @@ def revoke(self):
                 raise BoxOAuthException(network_response.status_code, network_response.content, url, 'POST')
             self._store_tokens(None, None)
 
+    def downscope_token(self, scopes, item=None, additional_data=None):
+        """
+        Get a downscoped token for the provided file or folder with the provided scopes
+
+        :param scope:
+            The scope(s) to apply to the resulting token
+        :type scopes:
+            `list` of :class:`TokenScope`
+        :param item:
+            The file or folder to get a downscoped token for
+        :type item:
+            :class:`Item`
+        :param additional_data:
+            Optional key value pairs which can be used to add/update the default data values in the request
+        :type additional_data:
+            `dict`
+        :return:
+            The downscoped token
+        :rtype:
+            `unicode`
+        """
+        self._check_closed()
+        with self._refresh_lock:
+            self._check_closed()
+            url = '{base_auth_url}/token'.format(base_auth_url=API.OAUTH2_API_URL)
```

Comment:
Rather than re-implementing most of `send_token_request`, can we refactor part of the definition of `send_token_request` into a new (private) method, which makes the API call but doesn't call `_store_tokens`? And then this method and `send_token_request` can both call that method.

In the future that could even be a public method, but it should probably be private to start with.

Reply:
I was actually thinking we should call the new method `_send_token_request` and rename `send_token_request` -> `send_token_request_and_store_tokens`

Reply:
Good idea, done